projects/components/src/table/cells/table-cell-csv-generator-lookup.service.ts

```diff
@@ -0,0 +1,49 @@
+import { Injectable, Type } from '@angular/core';
+import { TableCellCsvGeneratorBase } from './table-cell-csv-generator-base';
+import { CoreTableCellCsvGeneratorType } from './types/core-table-cell-csv-generator-type';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class TableCellCsvGeneratorLookupService {
+  private readonly csvGenerator: Map<string, TableCellCsvGeneratorConstructor<unknown, unknown>> = new Map();
+
+  public register(...csvGenerators: TableCellCsvGeneratorConstructor<unknown, unknown>[]): void {
+    csvGenerators.forEach(csvGenerator => {
+      this.csvGenerator.set(csvGenerator.type, csvGenerator);
+    });
+  }
+
+  public lookup<C, R>(type: string): TableCellCsvGeneratorConstructor<C, R> {
+    if (!this.csvGenerator.has(type)) {
+      throw Error(`Table cell csv generator of type '${type}' not registered.`);
+    }
+
+    return this.csvGenerator.get(type)! as TableCellCsvGeneratorConstructor<C, R>;
+  }
+}
+
+/*
+ * TableCellRendererConstructor is used by lookup service to dynamically instantiate cell CsvGenerators
+ */
+export interface TableCellCsvGeneratorConstructor<TCellData, TRowValue>
+  extends Type<TableCellCsvGeneratorBase<TCellData, TRowValue>>,
+    TableCellCsvGeneratorMetadata {}
+
+/*
+ * @TableCellCsvGenerator decorator is used to configure type for cell CsvGenerators
+ */
+
+export function TableCellCsvGenerator(
+  tableCellCsvGeneratorMetadata: TableCellCsvGeneratorMetadata
+): TableCellCsvGeneratorDecorator {
+  return (constructor: TableCellCsvGeneratorConstructor<unknown, unknown>): void => {
+    constructor.type = tableCellCsvGeneratorMetadata.type;
```

Comment:
do you mean something like a single service that contains all the column to csv mapper thats injected via a token, and if its available, let the table directly consume it?

Reply:
Reading my comment further, the second point I was making was actually that we can make the generators themselves injectable (similar to how our gql handlers are) and forgo the decorators at all. This has several benefits
- reduce registration complexity
- simplifies and optimizes consumption. A single root injectable instance rather than new'ing an instance of the generator for every usage.
- allows generators to inject like any other service.

Reply:
Updated this to reduce registration complexity.